src/energyplus/ForwardTranslator.cpp

```diff
@@ -979,8 +979,12 @@ namespace energyplus {
         break;
       }
       case openstudio::IddObjectType::OS_Coil_Cooling_DX: {
-        model::CoilCoolingDX dx = modelObject.cast<CoilCoolingDX>();
-        retVal = translateCoilCoolingDX(dx);
+        model::CoilCoolingDX coil = modelObject.cast<CoilCoolingDX>();
+        if (this->isHVACComponentWithinUnitary(coil)) {
+          retVal = translateCoilCoolingDXWithoutUnitary(coil);
+        } else {
+          retVal = translateCoilCoolingDX(coil);
+        }
```

Comment:
Could you add some comments here especially in the FT par... (a walk-through of your changes)

I'm a bit puzzled here because it seem that before we never used to put the Coil:Cooling:DX inside a CoilSystem:Cooling:DX object? When do we bump into the second branch and `translateCoilCoolingDX` is called?

Reply:
`Coil:Cooling:DX` is now a valid cooling coil object type on `CoilSystem:Cooling:DX` (along with `Coil:Cooling:DX:SingleSpeed`, `Coil:Cooling:DX:VariableSpeed`, etc). So I'm following these existing cooling coil object types as a guide.

Looks like you'd jump into this new method when the `Coil:Cooling:DX` is not within either a `AirLoopHVAC:UnitarySystem`, CoilSystem:Cooling:DX:HeatExchangerAssisted`, `ZoneHVAC:PackagedTerminalAirConditioner`, or `ZoneHVAC:PackagedTerminalHeatPump`.

Reply:
Which is never currently, which was my point. https://github.com/NREL/OpenStudio/blob/8dc1ad4ca696c5b479a04aca3479c898001d96aa/src/model/CoilCoolingDX.cpp#L150-L153

I think your FT is actually right, and it's the model implementation that's wrong. We should be able to place this coil on an AirLoopHVAC branch, and at model time we wrap it in a CoilSystem:Cooling:DX. Same as Coil:Cooling:DX:SingleSpeed (and others)